main.cpp

```diff
@@ -25,33 +28,34 @@ int main(int argc, char **argv)
 
 	int stringLen = strlen(PortNumParam);
 	// Check for command line parameters
-	for (int i = 0; i < argc; i++)
+	for (int i = 0; i < argc; ++i)
 	{
 		if (strncmp(argv[i], PortNumParam, stringLen) == 0)
 		{
 			portNum = atoi(&argv[i][stringLen]);
 		}
 	}
 
-	GameServer gameServer;
-	int errorCode = gameServer.StartServer(portNum);
-
-	if (errorCode != 0)
-	{
-		LogMessage("Game Server failed to start");
-		return errorCode;
+	std::unique_ptr<GameServer> gameServer;

```

Comment:
I didn't see a good way to instantiate GameServer without making it a pointer, although I didn't want to make it a pointer.